File: 0215.Kth Largest Element in an Array.cs


var nums = new int[]{3,2,1,5,6,4};
var k = 2;
System.Console.WriteLine(new Solution().FindKthLargest(nums,k));

public class Solution
{
    public int FindKthLargest(int[] nums, int k)
    {
        PriorityQueue<int, int> pq = new();
        foreach (var n in nums)
        {
            pq.Enqueue(n,n);
            if(pq.Count>k){
                pq.Dequeue();
            }
        }
        return pq.Dequeue();
    }
}

File: 0215.Kth Largest Element in an Array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(k);
        for (int num : nums) {
            if (pq.size() < k) {
                pq.offer(num);
            }else if(pq.peek() < num){
                pq.poll();
                pq.offer(num);
            }
        }
        return pq.poll();
    }
}
